Updated Airflow guides based on ebook publication (#232)
astronomer/airflow-guides: guides/what-is-a-hook.md

Operators are the main building blocks of Airflow, but operators rely heavily upon Hooks to interact with all of their source and destination systems.

> Hooks are used as a way to abstract the methods you would use against a source system, but do not contain the logic for how that system is interacted with. Hooks should always be used to connect with any external system.

Reviewer: ```suggestion
Hooks are used as a way to abstract the methods you would use against a source system, but often contain limited logic for how that system is interacted with. Hooks should always be used to connect with any external system.
```

@jwitz reworded this a bit because it's not _entirely_ correct as is. Hooks sometimes contain functionality for actually "doing things" in addition to just connecting to the source system. For example, the S3Hook has methods to load files to S3, delete files, list keys, etc.
